polyply/src/build_file_parser.py

```diff
@@ -258,8 +261,53 @@ def _base_parser_geometry(tokens, _type):
 
         parameters.append(_type)
         geometry_def["parameters"] = parameters
+
+        is_good_geometry = _check_geometry_def(geometry_def, _type)
+        if not is_good_geometry:
+            msg = ("Geometry restriction {_type} {resname} {start} {stop} "
+                   "extends beyond definite positive coordinates relative "
+                   "to the center point {x:.3f} {y:.3f} {z:.3f}. Be aware "
+                   "polyply only builds in positive coordinate space.")
+            LOGGER.warning(msg, _type=_type, x=point[0], y=point[1], z=point[2],
+                           resname=tokens[0], start=tokens[1], stop=tokens[2])
         return geometry_def
 
+def _check_geometry_def(geom_def, geom_type):
+    """
+    Raise a warning if the point of reference
+    for the geometry type is too close to the
+    origin.
+
+    Parameters
+    ----------
+    geom_def: dict
+        dict with entries: resname, start, stop, point, parameters
+    geom_type: str
+        one of sphere, cylinder, rectangle
+    """
+    msg = ("Geometry restriction {geom_def} extends beyond definite "
+           "positive coordinates relative to the center point "
+           "{x:.3f} {y:.3f} {z:.3f}. Be aware polyply only builds "
+           "in positive coordinate space.")
+
+    point = geom_def['parameters'][1]
+    if geom_type == "sphere":
+        radius = geom_def['parameters'][2]
+        if any( i >= j for i, j in zip([radius, radius, radius], point)):
```

Comment:
Is checking the cardinal axes enough? Or you do need to do `(point**2).sum()` vs `3*radius**2`?